### ISSUE TITLE
OSX Installation Without Xcode?

### ISSUE THREAD

**Issue by bakercp** [olab-io/ofSketch#134]
Ideally we would install nothing to get running, but alas we usually have to install Xcode.

Xcode from the app store is the easiest way to get going.

The latest command line tools [command_line_tools_for_osx_mountain_lion_april_2014.dmg](http://adcdownload.apple.com/Developer_Tools/command_line_tools_os_x_mountain_lion_for_xcode__april_2014/command_line_tools_for_osx_mountain_lion_april_2014.dmg) for 10.8 ~~do not contain any SDK bundles so it is required to fully install Xcode.~~  They do actually -- it's just the system Root `/`

To Do:
- [ ] Is it possible to install with only command line tools on 10.9? [command_line_tools_for_osx_10.9_september_2014.dmg](http://adcdownload.apple.com/Developer_Tools/command_line_tools_os_x_10.9_for_xcode__september_2014/command_line_tools_for_osx_10.9_september_2014.dmg)

![cursor_and_downloads](https://cloud.githubusercontent.com/assets/300484/4635103/69885da0-53d6-11e4-929c-28f850c8ad20.png)

**Comment by bakercp** [comment]
I figured it out ... with some makefile modifications I just successfully ran ofSketch with ONLY command line tools installed.  !!

The key is that the "sysroot" is actually just the root directory and the "/"

I'm updating makefiles and we should be able to avoid much pain getting started now ....  

testing on 10.9.

@brannondorsey @mikewesthad

**Comment by mikewesthad** [comment]
// todo: finish making time machine :)

**Comment by bakercp** [comment]
OK.  Did a lot of research on the absolute easiest way to get up and running on OSX.  On 10.9, the easiest way will be to have ofSketch launch an install script with `xcode-select --install` if it doesn't detect command line tools.

On 10.8, it seems that the only legit way to do it is for the user to sign into the apple developer site, download command line tools and run it.

Unfortunately the 10.8 version of xcode-select does not have the --install flag, which hooks into Apple's software update CDN.  I looked deep into the software update route and it seems that there are no developer tools available via that route for 10.8.

I want to look into it to see if we can bundle command line tools with ofSketch, but I kind of doubt it ...

Here's a nice summary:

http://www.moncefbelyamani.com/how-to-install-xcode-homebrew-git-rvm-ruby-on-mac/#clt-ml

**Comment by mikewesthad** [comment]
I guess the computers in the 416 are running 10.8.  I remember trying to run `xcode-select --install` on one machine, and it didn't recognize the flag.

**Comment by bakercp** [comment]
Yeah, that is probably right.  Although I swear at least one of them was
running 10.9.5 ...

---

http://christopherbaker.net

**Comment by bigfatbrowncat** [comment]
Just in case: there is no absolute necessity to install xcode toolkit to compile C++. You may, instead, use brew (for instance). Just like in Windows you can use mingw-w64. And if you are going to make user code really cross-platform, I'd recommend you to consider using a single toolchain (gcc) everywhere. Or even provide your own version of it (in advance).
